share/spack/gitlab/cloud_pipelines/configs/ci.yaml

```diff
@@ -0,0 +1,51 @@
+ci:
+  target: gitlab
+
+  broken-tests-packages:
+    - gptune
+
+  broken-specs-url: "https://dummy.io" # s3://spack-binaries/broken-specs"
+
+  pipeline-gen:
+  - build-job:
+      script::
+      - - uname -a || true
+        - grep -E "vendor|model name" /proc/cpuinfo 2>/dev/null | sort -u || head -n10 /proc/cpuinfo 2>/dev/null || true
+        - nproc
+        - curl -Lfs "https://github.com/JuliaBinaryWrappers/GNUMake_jll.jl/releases/download/GNUMake-v4.3.0+1/GNUMake.v4.3.0.x86_64-linux-gnu.tar.gz" -o gmake.tar.gz
+        - printf "fef1f59e56d2d11e6d700ba22d3444b6e583c663d6883fd0a4f63ab8bd280f0f gmake.tar.gz" | sha256sum --check --strict --quiet
+        - tar -xzf gmake.tar.gz -C /usr bin/make 2> /dev/null
+        - . "./share/spack/setup-env.sh"
+        - spack --version
+        - spack arch
+        - cd ${SPACK_CONCRETE_ENV_DIR}
+        - spack env activate --without-view .
+        - spack config add "config:install_tree:projections:${SPACK_JOB_SPEC_PKG_NAME}:'morepadding/{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'"
+        - mkdir -p ${SPACK_ARTIFACTS_ROOT}/user_data
+        - if [[ -r /mnt/key/intermediate_ci_signing_key.gpg ]]; then spack gpg trust /mnt/key/intermediate_ci_signing_key.gpg; fi
```

Comment:
Since you opened this PR, some things changed in the stacks on `develop`, and it won't be easy to know what might be missing, since things got moved all around.  One thing I know is missing for sure is support for UO doing protected builds and doing signing, see e.g. [here](https://github.com/spack/spack/blob/develop/share/spack/gitlab/cloud_pipelines/stacks/e4s/spack.yaml#L253-L256).  Can you add that functionality here and maybe take a quick look to see if you can spot any other differences between the script there and here?